src/data_gradients/datasets/detection/coco_format_detection_dataset.py

```diff
@@ -0,0 +1,105 @@
+
+    #### Expected folder structure
+    The dataset folder structure should include at least one sub-directory for images and one JSON file for annotations.
+
+    Example:
+    ```
+        dataset_root/
+            ├── images/
+            │   ├── train/
+            │   │   ├── 1.jpg
+            │   │   ├── 2.jpg
+            │   │   └── ...
+            │   ├── test/
+            │   │   ├── ...
+            │   └── validation/
+            │       ├── ...
+            └── annotations/
+                ├── train.json
+                ├── test.json
+                └── validation.json
+    ```
+    #### Expected Annotation File Structure
+    The annotation files must be structured in JSON format following the COCO data format.
+
+    #### Instantiation
+    ```
+    dataset_root/
+        ├── images/
+        │   ├── train/
+        │   │   ├── 1.jpg
+        │   │   ├── 2.jpg
+        │   │   └── ...
+        │   ├── val/
+        │   │   ├── ...
+        │   └── test/
+        │       ├── ...
+        └── annotations/
+            ├── train.json
+            ├── test.json
+            └── validation.json
+    ```
+
+    ```python
+    from data_gradients.datasets.detection import COCOFormatDetectionDataset
+
+    train_set = COCOFormatDetectionDataset(
+        root_dir="<path/to/dataset_root>", images_subdir="images/train", annotation_file_path="annotations/train.json"
+    )
+    val_set = COCOFormatDetectionDataset(
+        root_dir="<path/to/dataset_root>", images_subdir="images/validation", annotation_file_path="annotations/validation.json"
+    )
+    ```
+    """
+
+    def __init__(self, root_dir: str, images_subdir: str, annotation_file_path: str):
+        """
+        :param root_dir:                Where the data is stored.
+        :param images_subdir:           Local path to directory that includes all the images. Path relative to `root_dir`.
+        :param annotation_file_path:    Local path to annotation file. Path relative to `root_dir`.
+        """
+
+        self.base_dataset = CocoDetection(
+            root=os.path.join(root_dir, images_subdir),
+            annFile=os.path.join(root_dir, annotation_file_path),
+        )
+
+        self.class_ids = self.base_dataset.coco.getCatIds()
+
+        categories = self.base_dataset.coco.loadCats(self.class_ids)
+        self.class_names = [category["name"] for category in categories]
+
+    def load_image(self, index: int) -> np.ndarray:
+        image = self.base_dataset[index][0]
+        return np.array(image)
+
+    def load_labels(self, index: int) -> np.ndarray:
+        annotations = self.base_dataset[index][1]
+
+        labels = []
+        for annotation in annotations:
+            class_id = annotation["category_id"]
+            x, y, w, h = annotation["bbox"]
+            labels.append((int(class_id), float(x), float(y), float(w), float(h)))
+        return np.array(labels)
```

Comment:
You may want to write something like `np.array(labels, dtype=np.float32).reshape(-1, 5)` to ensure even for empty sample the labels you return from this function will be 2-dimensional array. And explicit dtype is also necessary from the same considerations - to guarantee consistency for empty & non-empty samples